### PR TITLE
splitup rviz display into its own package

### PR DESCRIPTION
**Pull request by LeroyR**
follow up to #158

### CODE REVIEW

**Review by SteveMacenski — COMMENTED** [review]
Part of the issue I see here is that if someone installs slam toolbox, and the rviz plugins don’t appear I’m going to get flooded with tickets and ros answer questions. 

Can you add this as a dependency of the main package? I believe in catkin you can ignore dep keys if you don’t want it installed.

**Comment by LeroyR** [comment]
Not having default dependency to rviz is the whole point of this.

Our build tool does not allow that and it seems like a major hassle to create a special dep ignore rule for this package.

To include the stuff by default you could use a metapackage named slam_toolbox :smiley: 
Also if this is in some default for ros2 also add the slam_toolbox_rviz package there.
or just add this as rviz dep :walking_man:

**Comment by SteveMacenski** [comment]
> Not having default dependency to rviz is the whole point of this.

Yeah. :man_facepalming: I know. However a great out of box experience is something I strive for. I'm not sure what the best option is, thinking about this again, a metapackage could solve this. I hadn't understood in your original PR that you wanted to lower the image sizes. You made a good argument for `msgs` and I thought you just tacked on the rviz plugins for good measure. I think we missed each other in communication.

> it seems like a major hassle to create a special dep ignore rule for this package.

it doesn't seem like that much hassle to me, if this is all just in a script for deploying on your systems, adding a flag shouldn't be a big deal. I do understand the mentality of not wanting snowflakes, but think about the frustration of other users that install this and expect it to just work out of the box? Lowering your image sizes for an actual deployment is a good thing to do, but with that comes a little bit of extra specialization. 

> Also if this is in some default for ros2 also add the slam_toolbox_rviz package there. or just add this as rviz dep

I don't quite understand. Can you rephrase? 

---

Let me sit on this for a day and see what I think. I think I could be convinced for a metapackage. If we were to make a metapackage, we cannot change the name of slam_toolbox. That will break everyone’s launch and run scripts. We will need to make the metapackage name not be slam_toolbox. The _core will need to stay slam _toolbox. 

This will still break anyone’s saved rviz configs, but that’s a much smaller deal.

**Comment by LeroyR** [comment]
> [..]  I do understand the mentality of not wanting snowflakes, but think about the frustration of other users that install this and expect it to just work out of the box? Lowering your image sizes for an actual deployment is a good thing to do, but with that comes a little bit of extra specialization.

Yes if we want to get as small as possible. BUT i think apt install ros-navigation on a minimal robot image should not default to qt installation. 

> 
> > Also if this is in some default for ros2 also add the slam_toolbox_rviz package there. or just add this as rviz dep
> 
> I don't quite understand. Can you rephrase?

I expect it to be part of metapackages e.g. ros-navigation and desktop-full, so we can just add slam_toolbox_rviz to desktop-full for the good default UX.

Another (semi-joke) option is to just add this as dependency to rviz so it it there as soon as someone installs rviz.

**Comment by SteveMacenski** [comment]
OK, I think a metapackage here makes sense.

I think the core package should remain as `slam_toolbox` but the metapackage can be called something else. That way we don't break launch files. 

As for the messages / rviz, I think we should create a new Noetic branch that these will apply to. Since theyre ABI breaking changes, I'm not sure they should be deployed with new debians in melodic.  Given C++ hasn't really changed since Indigo, the noetic branch could be used for your melodic needs (the way that many people use the melodic branch for Kinetic) 

Thoughts?

**Comment by SteveMacenski** [comment]
Thoughts?

**Comment by LeroyR** [comment]
Sorry - I was out for 2 weeks.

Switching to another branch would be fine to. Still, the functionality of the slam_toolbox package changes (no display) so it _could_ still cause migration issues e.g. next LTS upgrade - if people simple sed melodic/Oxx on the package list and we still have to note that somewhere. 

Metapackages for installs always make sense from my pov.

**Comment by SteveMacenski** [comment]
No worries. 

So retarget the PR to `noetic-devel`? Its changes for migration between distributions, but that's fine. We are allowed to change things then.

**Review by SteveMacenski — APPROVED** [review]

**Comment by SteveMacenski** [comment]
Whoops, sorry, this slipped my mind. Looks good, merging.
